prow/plugins/releasenote/releasenote.go

```diff
@@ -29,98 +31,272 @@ import (
 const pluginName = "release-note"
 
 const (
-	releaseNoteActionRequired        = "release-note-action-required"
-	releaseNoteNone                  = "release-note-none"
+	// deprecatedReleaseNoteLabelNeeded is the previous version of the
+	// releaseNotLabelNeeded label, which we continue to honor for the
+	// time being
 	deprecatedReleaseNoteLabelNeeded = "release-note-label-needed"
-	releaseNoteLabelNeeded           = "do-not-merge/release-note-label-needed"
-	releaseNote                      = "release-note"
+
+	releaseNoteLabelNeeded    = "do-not-merge/release-note-label-needed"
+	releaseNote               = "release-note"
+	releaseNoteNone           = "release-note-none"
+	releaseNoteActionRequired = "release-note-action-required"
+
+	releaseNoteFormat       = `Adding %s because the release note process has not been followed.`
+	releaseNoteSuffixFormat = `One of the following labels is required %q, %q, or %q.
+Please see: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#write-release-notes-if-needed.`
+	parentReleaseNoteFormat = `All 'parent' PRs of a cherry-pick PR must have one of the %q or %q labels, or this PR must follow the standard/parent release note labeling requirement.`
+
+	noReleaseNoteComment = "none"
+	actionRequiredNote   = "action required"
 )
 
 var (
+	releaseNoteSuffix         = fmt.Sprintf(releaseNoteSuffixFormat, releaseNote, releaseNoteActionRequired, releaseNoteNone)
+	releaseNoteBody           = fmt.Sprintf(releaseNoteFormat, releaseNoteLabelNeeded)
+	deprecatedReleaseNoteBody = fmt.Sprintf(releaseNoteFormat, deprecatedReleaseNoteLabelNeeded, releaseNoteSuffix)
+	parentReleaseNoteBody     = fmt.Sprintf(parentReleaseNoteFormat, releaseNote, releaseNoteActionRequired)
+
+	noteMatcherRE = regexp.MustCompile(`(?s)(?:Release note\*\*:\s*(?:<!--[^<>]*-->\s*)?` + "```(?:release-note)?|```release-note)(.+?)```")
+	cpRe          = regexp.MustCompile(`Cherry pick of #([[:digit:]]+) on release-([[:digit:]]+\.[[:digit:]]+).`)
+
 	allRNLabels = []string{
 		releaseNoteNone,
 		releaseNoteActionRequired,
 		deprecatedReleaseNoteLabelNeeded,
 		releaseNoteLabelNeeded,
 		releaseNote,
 	}
-
-	releaseNoteRe               = regexp.MustCompile(`(?mi)^/release-note\s*$`)
-	releaseNoteNoneRe           = regexp.MustCompile(`(?mi)^/release-note-none\s*$`)
-	releaseNoteActionRequiredRe = regexp.MustCompile(`(?mi)^/release-note-action-required\s*$`)
 )
 
 func init() {
-	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest)
```

Comment:
Is it possible to use issue events instead of PR events? IIRC issue events have the labels in the webhook, so we would save some API tokens.

Reply:
Issue events would include the labels, but they wouldn't include the PR itself which is needed to check the branch that the PR is for. So switching to issue event instead of PR events would use the same number of API tokens.